src/algorithms/ctmrg.jl

```diff
@@ -1,24 +1,46 @@
+# TODO: add option for different projector styles (half-infinite, full-infinite, etc.)
+"""
+    struct ProjectorAlg{S}(; svd_alg = TensorKit.SVD(), trscheme = TensorKit.notrunc(),
+                           fixedspace = false, verbosity = 0)
+
+Algorithm struct collecting all projector related parameters. The truncation scheme has to be
+a `TensorKit.TruncationScheme`, and some SVD algorithms might have further restrictions on what
+kind of truncation scheme can be used. If `fixedspace` is true, the truncation scheme is set to
+`truncspace(V)` where `V` is the environment bond space, adjusted to the corresponding
+environment direction/unit cell entry.
+"""
+@kwdef struct ProjectorAlg{S}
+    svd_alg::S = TensorKit.SVD()
+    trscheme::TruncationScheme = TensorKit.notrunc()
+    fixedspace::Bool = false
+    verbosity::Int = 0
+end
+
+function fix_spaces(alg::ProjectorAlg)
+    return ProjectorAlg(;
+        svd_alg=alg.svd_alg, trscheme=alg.trscheme, fixedspace=true, verbosity=alg.verbosity
+    )
+end
```

Comment:
I think we already have Accessors.jl in our dependencies. This then should be equivalent to 
```julia
alg_fixed = @set alg.fixedspace = true
```

Reply:
I didn't know about Accessors.jl yet, that's super helpful. In that case, the `fix_spaces` function is not really needed anymore!